geco/generator.py

```diff
@@ -17,3 +18,41 @@ def __next__(self):
 
     def seed(self, seed: int):
         self.rng.seed(seed)
+
+
+@py_random_state(-1)
+def common_substructure_generator(
+    instance_generation_function,
+    instance_params_generation_function,
+    base_params,
+    new_params,
+    expand_params_function,
+    seed=0,
+):
+    """
+    Generates instances that have common substructure
+
+    Parameters
+    ----------
+    instance_generation_function:
+        base function that defines MIP
+    instance_params_generation_function:
+        function to generate parameters for instance_generation_function
+    base_params:
+        parameters for the common substructure
+    new_params:
+        parameters for the required expanded instance
+    expand_params_function:
+        function to expand instance params using new_params
+    seed: int, random object or None
+        for randomization
+
+    Returns
+    -------
+        generator object
+    """
+    while True:
+        base_result = instance_params_generation_function(*base_params, seed=seed)
+        yield instance_generation_function(
+            *expand_params_function(new_params, base_result, seed=seed)
```

Comment:
If I'm not mistaken, this will always result in the same expansion? I think the generator needs an internal random device like the basic generator has.

Reply:
I'm wrong.